wrangler/src/txFunctions/run.js

```diff
@@ -168,45 +96,40 @@ export default async ({ request, params, env, ctx }) => {
   .then(async (res) => {
     watch.mark('Ran txFunction')
 
-    const now = moment.utc().format('x')
     const cost = new BigNumber(watch.getTotalTime()).dividedBy(RUN_DIVISOR).toFixed(7)
-    const feeTotal = feeTotalBigNumber.toFixed(7)
-    const feeSpent = feeSpentBigNumber.plus(cost).toFixed(7)
+    const feeBalanceRemaining = feeBalance.minus(cost).toFixed(7)
 
-    await TX_FEES.put(claimableBalanceId, 'OK', {metadata: {
-      date: now,
-      total: feeTotal,
-      spent: feeSpent
+    await TX_FEES.put(authedPublicKey, 'OK', {metadata: {
+      lastModifiedTime: moment.utc().format('x'),
+      balance: feeBalanceRemaining
```

Comment:
Do we allow the negative `balance` here? Because right now there's no check for this.

Reply:
Yup.

Currently, the Turret only rejects a user if they have 0 or less fee balance. This was in hopes to avoid setting consistent turret fee, and with the assumption `XLM_FEE_MIN` is large enough to avoid this getting abused.

As such, a user can technically accrue some small negative balance that they will pay back once they submit another fee payment.